keyboards/rmkeebs/rm_numpad/rm_numpad.h

```diff
@@ -0,0 +1,62 @@
+/* Copyright 2021 RuckerMachine
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "quantum.h"
+
+/* This is a shortcut to help you visually see your layout.
+ *
+ * The first section contains all of the arguments representing the physical
+ * layout of the board and position of the keys.
+ *
+ * The second converts the arguments into a two-dimensional array which
+ * represents the switch matrix.
+ */
+
+
+/* rm_numpad default layout
+ * .-------------------.
+ * |PAUS|PREV|NEXT|MUTE|
+ * |-------------------|
+ * |NLCK|   /|   *|   -|
+ * |-------------------|
+ * |   7|   8|   9|    |
+ * |--------------|    |
+ * |   4|   5|   6|   +|
+ * |-------------------|
+ * |   1|   2|   3|    |
+ * |--------------|    |
+ * |        0|   .| Ent|
+ * '-------------------'
+ */
+
+
+#define LAYOUT( \
+    k00, k01, k02, k03,      \
+    k10, k11, k12, k13,      \
+    k20, k21, k22, k23, k24, \
+    k30, k31, k32, k33,      \
+    k40, k41, k42, k43, k44,  \
+    k50, k51, k52, k53, k54  \
+) { \
+    { k00, k01, k02, k03      },  \
+    { k10, k11, k12, k13      },  \
+    { k20, k21, k22, k23, k24 },  \
+    { k30, k31, k32, k33      },  \
+    { k40, k41, k42, k43, k44 },  \
+    { k50, k51, k52, k53, k54 }   \
+}	
```

Comment:
This layout macro is not correct. 

ALso, this looks to be a 6x4 numpad.  Please conform to the community layout:
https://github.com/drashna/qmk_firmware/blob/c53b9588c4a2b20c9dd059729ed98cd13412738d/layouts/default/numpad_6x4/default_numpad_6x4/keymap.c#L19

You can have additional layouts defined, as well.  In fact, you appear to do so in your info.json, but lack the actual layout macros here (which will cause them to not work)

Reply:
Sorry, I copied this format from another numpad a friend built.  I was under the assumption that the layout here reflects how the PCB is wired, and not a specific keymap. (I can provide the schematic if that's needed.)
I guess I'm confused about the difference between the layout listed here vs the layout listed in keymap.c

Reply:
The bottom half is how it is wired, as it's the electrical matrix.

The top half can be ... well, in any config you want.  The entire point of the layout macro here, is to allow you to remap things to be more human friendly. 

A good example is the planck rev6's layout, as its matrix is square, but the switch layout is not.

Reply:
Okay, I think I understand now.  So for the default layout I have here, it would be something like this:
```
#define LAYOUT_default( \
    k00, k01, k02, k03,      \
    k10, k11, k12, k13,      \
    k20, k21, k22, k24,      \
    k30, k31, k32,              \
    k40, k41, k42, k44,      \
        k51,     k53               \
) { \
    { k00, k01, k02, k03      },  \
    { k10, k11, k12, k13      },  \
    { k20, k21, k22, k23, k24 },  \
    { k30, k31, k32, k33      },  \
    { k40, k41, k42, k43, k44 },  \
    { k50, k51, k52, k53, k54 }   \
}	
```
Then similar ones for the other layouts, ortho and split plus.
Does this look right?

Reply:
Yup, that looks about right.

But definitely double check that this compiles and works right.

Also, it's name should be `LAYOUT_numpad_6x4`.   And this allows you to add `LAYOUTS = numpad_6x4` to your rules.mk, and allow for community layout support (eg, having one keymap that will compile on multiple devices).  
https://docs.qmk.fm/#/feature_layouts

Reply:
Okay, working on this now, I've read through everything you've linked a couple times, and the part about community layout support is causing more confusion.
The keymap in `/layouts/default/numpad_6x4` is not the same as the default keymap I have defined for my board.  Adding it like you're describing, would that overwrite the default keymap I'm trying to use?  I'm not sure I understand why I'd do this, they keymap in that default layout doesn't make much sense on my board.
I also see one in `/layouts/default/ortho_6x4`, and it's the same story there.

Reply:
@drashna I'm waiting on an answer to the above questions before I push any more changes.

Reply:
@drashna the changes you described above only break the functionality of this board.  While the keycap layout may be 6x4, the matrix is NOT 6x4.  I spent several hours trying to configure it the way you described, but the way I had it originally is the only way the keymaps I have defined will work.  Please approve and merge.

Reply:
The layout macro doesn't have to match the matrix layout.  That's actually the point.  It remaps the physical matrix into something that is more representative of the key layout per a human perspective. 

For instance, in the default keymap, you have a bunch of "KC_NO"s in layout, which ... shouldn't be there, really, if there are no keys there. 

Eg, this:

```c
    LAYOUT(
        KC_MPLY, KC_MPRV, KC_MNXT, KC_MUTE,
        KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS, 
        KC_P7,   KC_P8,   KC_P9,   KC_NO,   KC_PPLS,
        KC_P4,   KC_P5,   KC_P6,   KC_NO, 
        KC_P1,   KC_P2,   KC_P3,   KC_NO,   KC_PENT,
        KC_NO,   KC_P0,   KC_NO,   KC_PDOT, KC_NO
```

should look like this:
```c
    LAYOUT(
        KC_MPLY, KC_MPRV, KC_MNXT, KC_MUTE,
        KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS, 
        KC_P7,   KC_P8,   KC_P9,   
        KC_P4,   KC_P5,   KC_P6,   KC_PPLS,
        KC_P1,   KC_P2,   KC_P3,   
               KC_P0,     KC_PDOT, KC_PENT,
```

Assuming it looks like a normal 6x4 number pad. Which ... it does, according to the image. 
https://i.imgur.com/APQGsvU.png



That means that you'd need to reformat the layout macro to accommodate for that. 

```c

define ___ KC_NO

#define LAYOUT_numpad_6x4( \
    k00, k01, k02, k03, \
    k10, k11, k12, k13, \
    k20, k21, k22,      \
    k30, k31, k32, k24, \
    k40, k41, k42,      \
       k51,   k53, k44  \
) { \
    { k00, k01, k02, k03, ___ },  \
    { k10, k11, k12, k13, ___ },  \
    { k20, k21, k22, k23, ___ },  \
    { k30, k31, k32, ___, ___ },  \
    { k40, k41, k42, ___, k44 },  \
    { ___, k51, k52, k53, ___ }   \
}

#define LAYOUT_ortho_6x4( \
    k00, k01, k02, k03, k04, \
    k10, k11, k12, k13, k14, \
    k20, k21, k22, k23, k24, \
    k30, k31, k32, k33, k34, \
    k40, k41, k42, k43, k44, \
    k50, k51, k52, k53, k54  \
) { \
    { k00, k01, k02, k03, k04 },  \
    { k10, k11, k12, k13, k14 },  \
    { k20, k21, k22, k23, k24 },  \
    { k30, k31, k32, k33, k34 },  \
    { k40, k41, k42, k43, k44 },  \
    { k50, k51, k52, k53, k54 }   \
}
```

Reply:
Thank you for the clarification.  I've got everything up to date and tested.
I was never able to find in the documentation where it described defining underscores as KC_NO in the .h file as opposed to the keymap file.  If you are able to point me in the right direction so I can make sure I get it right for next time, I'd appreciate it.

Thanks again for your help.

Reply:
The define is mostly to make it easier to read.  It's not needed, but it's definitely easier to see what it going on.